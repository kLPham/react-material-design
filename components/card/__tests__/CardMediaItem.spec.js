import React from 'react';
import { shallow } from 'enzyme';
import CardMediaItem from '../cardMediaItem';

describe('<CardMediaItem />', () => {
    it('should render', () => {
        const imgSrc = 'http://google.com/image.jpeg';
        const cardMediaItem = shallow(
            <CardMediaItem imgSrc={imgSrc} />,
    );
        expect(cardMediaItem).toMatchSnapshot();
    });
    it('should render with a title', () => {
        const imgSrc = 'http://google.com/image.jpeg';
        const title = 'this is my title';
        const cardMediaItem = shallow(
            <CardMediaItem imgSrc={imgSrc} title={title} />,
    );
        expect(cardMediaItem).toMatchSnapshot();
    });
    it('should render with imgSize', () => {
        const imgSrc = 'http://google.com/image.jpeg';
        const imgSize = 1.5;
        const cardMediaItem = shallow(
            <CardMediaItem imgSrc={imgSrc} imgSize={imgSize} />,
    );
        expect(cardMediaItem).toMatchSnapshot();
    });
    it('should render with children', () => {
        const imgSrc = 'http://google.com/image.jpeg';
        const cardMediaItem = shallow(
            <CardMediaItem imgSrc={imgSrc}>Text</CardMediaItem>,
    );
        expect(cardMediaItem).toMatchSnapshot();
    });
});
