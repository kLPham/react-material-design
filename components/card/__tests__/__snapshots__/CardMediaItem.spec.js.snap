// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardMediaItem /> should render 1`] = `
<div
  className="mdc-card__horizontal-block"
>
  <img
    alt=""
    className="mdc-card__media-item"
    src="http://google.com/image.jpeg"
  />
</div>
`;

exports[`<CardMediaItem /> should render with a title 1`] = `
<div
  className="mdc-card__horizontal-block"
>
  <CardTitle
    larger={false}
    subtitle=""
    title="this is my title"
  />
  <img
    alt=""
    className="mdc-card__media-item"
    src="http://google.com/image.jpeg"
  />
</div>
`;

exports[`<CardMediaItem /> should render with children 1`] = `
<div
  className="mdc-card__horizontal-block"
>
  <img
    alt=""
    className="mdc-card__media-item"
    src="http://google.com/image.jpeg"
  />
  Text
</div>
`;

exports[`<CardMediaItem /> should render with imgSize 1`] = `
<div
  className="mdc-card__horizontal-block"
>
  <img
    alt=""
    className="mdc-card__media-item mdc-card__media-item--1dot5x"
    src="http://google.com/image.jpeg"
  />
</div>
`;
